cluster/terraform_kubernetes/alertmanager.tf

```diff
@@ -132,3 +132,54 @@ resource "kubernetes_service" "alertmanager" {
     type = "NodePort"
   }
 }
+
+
+
+data "azurerm_key_vault_secret" "alertmanager_auth" {
+  name         = "ALERTMANAGER-AUTH"
+  key_vault_id = data.azurerm_key_vault.key_vault.id
+}
+
+
+resource "kubernetes_secret" "alertmanager_basic_auth" {
+  metadata {
+    name      = "alertmanager-basic-auth"
+    namespace = kubernetes_namespace.default_list["monitoring"].metadata[0].name
+  }
+
+  data = {
+    auth = data.azurerm_key_vault_secret.alertmanager_auth.value
+  }
+}
+
+resource "kubernetes_ingress_v1" "alertmanager_ingress" {
+  wait_for_load_balancer = true
+  metadata {
+    name      = "alertmanager"
+    namespace = "monitoring"
+    annotations = {
+      "nginx.ingress.kubernetes.io/auth-type"   = "basic"
+      "nginx.ingress.kubernetes.io/auth-secret" = "alertmanager-basic-auth"
```

Comment:
Can we reference the terraform resource directly to force the dependency? ie `kubernetes_secret.alertmanager_basic_auth.name`

Same for the other ingresses

Reply:
Added depends

Reply:
There is no need for the depends_on. Use:
```
"nginx.ingress.kubernetes.io/auth-secret" = kubernetes_secret.alertmanager_basic_auth.name
```

Reply:
fixed